src/MongoDB.Bson/IO/ElementAppendingBsonWriter.cs

```
@@ -51,6 +51,37 @@ internal sealed class ElementAppendingBsonWriter : WrappingBsonWriter
         }
 
         // public methods
+        public override void WriteRawBsonDocument(IByteBuffer slice)
+        {
+            WriteStartDocument();
+
+            if (Wrapped is BsonBinaryWriter binaryWriter)
+            {
+                // just copy the bytes (without the length and terminating null)
+                var lengthBytes = new byte[4];
+                slice.GetBytes(0, lengthBytes, 0, 4);
+                var length = BitConverter.ToInt32(lengthBytes, 0);
+                using (var elements = slice.GetSlice(4, length - 5))
+                {
+                    var stream = binaryWriter.BsonStream;
+                    stream.WriteSlice(elements);
+                }
+            }
+            else
+            {
+                // copy the elements themselves
```

This code path has not been tested. Probably we need to add a unit test for it.

---

Since the class is internal and we don't expect it to be used in this context, we could just throw an exception here.

---

@rstam what do you think?

---

I have no strong opinion. You can throw a `NotSupportedException` if you want.

---

I've added `NotSupportedException`